tests/unit/s2n_psk_test.c

```diff
@@ -194,9 +345,177 @@ int main(int argc, char **argv)
             S2N_BLOB_EXPECT_EQUAL(test_psk.early_secret, early_secret);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
         }
 
-        EXPECT_SUCCESS(s2n_connection_free(conn));
+        /* Test s2n_psk_write_binder with known values */
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+            DEFER_CLEANUP(struct s2n_psk psk, s2n_psk_free);
+            EXPECT_SUCCESS(s2n_psk_init(&psk, S2N_PSK_TYPE_RESUMPTION));
+            EXPECT_SUCCESS(s2n_psk_new_secret(&psk, resumption_secret.data, resumption_secret.size));
+
+            struct s2n_stuffer out;
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&out, 0));
+
+            EXPECT_OK(s2n_psk_write_binder(conn, &psk, &binder_hash, &out));
+
+            uint8_t binder_size;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint8(&out, &binder_size));
+            EXPECT_EQUAL(binder_size, s2n_stuffer_data_available(&out));
+            EXPECT_EQUAL(binder_size, finished_binder.size);
+
+            uint8_t *binder_data;
+            EXPECT_NOT_NULL(binder_data = s2n_stuffer_raw_read(&out, binder_size));
+            EXPECT_BYTEARRAY_EQUAL(binder_data, finished_binder.data, binder_size);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&out));
+        }
+
+        /* Test s2n_psk_write_binder_list with known values */
+        {
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+            struct s2n_psk *psk;
+            EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &psk));
+            EXPECT_SUCCESS(s2n_psk_init(psk, S2N_PSK_TYPE_RESUMPTION));
+            EXPECT_SUCCESS(s2n_psk_new_identity(psk, identity.data, identity.size));
+            EXPECT_SUCCESS(s2n_psk_new_secret(psk, resumption_secret.data, resumption_secret.size));
+
+            struct s2n_stuffer out;
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&out, 0));
+
+            EXPECT_OK(s2n_psk_write_binder_list(conn, &client_hello_prefix, &out));
+
+            uint16_t binder_list_size;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint16(&out, &binder_list_size));
+            EXPECT_EQUAL(binder_list_size, s2n_stuffer_data_available(&out));
+
+            uint8_t binder_size;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint8(&out, &binder_size));
+            EXPECT_EQUAL(binder_size, s2n_stuffer_data_available(&out));
+            EXPECT_EQUAL(binder_size, finished_binder.size);
+
+            uint8_t *binder_data;
+            EXPECT_NOT_NULL(binder_data = s2n_stuffer_raw_read(&out, binder_size));
+            EXPECT_BYTEARRAY_EQUAL(binder_data, finished_binder.data, binder_size);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&out));
+        }
+
+        /* Test s2n_psk_write_binder_list with multiple PSKs */
+        {
```

Comment:
It seems weird that duplicate psks/identities are allowed, but I guess there's nothing in the RFC against it...

Reply:
We might want to do some validation that customers can't load duplicates when we add the external API to configure PSKs, but that's out of scope of this change.